src/main/scala/com/cloud/apim/otoroshi/extensions/biscuit/entities/verifier.scala

```diff
@@ -190,42 +193,54 @@ case class VerifierConfig(
         .runFold(RemoteFactsData())(_.merge(_))(env.otoroshiMaterializer)
     }
 
-    remoteFactsF.map { remoteFacts =>
+    remoteFactsF.flatMap { remoteFacts =>
 
       remoteFacts.facts.foreach(f => verifier.add_fact(f))
       remoteFacts.checks.foreach(f => verifier.add_check(f))
       remoteFacts.roles.foreach(f => verifier.add_fact(f))
       remoteFacts.acl.foreach(f => verifier.add_fact(f))
 
-      // Check for token revocation
-      val ids = biscuitToken.revocation_identifiers().asScala.map(_.toHex).toList ++ remoteFacts.revoked
-      if (revokedIds.nonEmpty && ids.exists(revokedIds.contains)) {
-        Left(handleBiscuitErrors(new Error.FormatError.DeserializationError("Revoked token")))
-      } else {
-        val maxFacts = 1000 // TODO: from config
-        val maxIterations = 100 // TODO: from config
-        val maxTime = java.time.Duration.ofMillis(100) // TODO: from config
-        // Perform authorization
-        if (verifier.policies().isEmpty) {
-          Try(verifier.allow().authorize(new RunLimits(maxFacts, maxIterations, maxTime))).toEither match {
-            case Left(err: org.biscuitsec.biscuit.error.Error) =>
-              Left(handleBiscuitErrors(err))
-            case Left(err) =>
-              Left(handleBiscuitErrors(new org.biscuitsec.biscuit.error.Error.InternalError()))
-            case Right(_) =>
-              Right(())
-          }
-        } else {
-          Try(verifier.authorize(new RunLimits(maxFacts, maxIterations, maxTime))).toEither match {
-            case Left(err: org.biscuitsec.biscuit.error.Error) =>
-              Left(handleBiscuitErrors(err))
-            case Left(err) =>
-              // TODO: log
-              Left(handleBiscuitErrors(new org.biscuitsec.biscuit.error.Error.InternalError()))
-            case Right(_) =>
-              Right(())
+      val listOfTokenRevocationIds = biscuitToken.revocation_identifiers().asScala.map(_.toHex).toList
+      env.adminExtensions.extension[BiscuitExtension].get.datastores.biscuitRevocationDataStore.list().flatMap {
```

Comment:
We cannot use this mechanism in the routing reactor, we need to manually check all revocations identifiers of the token one by one against the revocation datastore